### PR TITLE
[SSDK-498] Update SwiftLint, SwiftFormat

### PR DESCRIPTION
**Pull request by aokj4ck**
### Description

Fixes [#498](https://mapbox.atlassian.net/browse/SSDK-498)

- Update SwiftLint 0.44 → 0.53 and rules
- Update Swiftformat → 0.52.11 and configuration
- Add Brewfile
- Update Build Phases for SwiftLint
- Update Carthage versions referenced in README and Makefile
- Update README for `brew bundle install` and setting up git hooks
- Add git pre-commit hook to format staged Swift files
- Update Sources, Examples, and Tests to fix some swiftlint violations

### Checklist
- [x] Update `CHANGELOG`

### CODE REVIEW

File: Examples/mapbox_access_token [hunk 1]
```diff
@@ -0,0 +1 @@
+pk.eyJ1IjoiamFsdG8yNDU2IiwiYSI6ImNscTVoN2VxMDBobDUycnFseGhueG5rNGsifQ.xiqQ9TeCvDxnkCjzgeXoyw
```
Comment (azarovalex) [inline Examples/mapbox_access_token:1]: 🫡

File: README.md [hunk 2]
```diff
@@ -53,18 +53,18 @@ The Search SDK is pre-configured for autocomplete, local search biasing, and inc
...
     2. File content:
-```
-machine api.mapbox.com
-login mapbox
-password sk.ey_your_access_token_wit_Read_permission
-```
+    ```
+    machine api.mapbox.com
+    login mapbox
+    password sk.ey_your_access_token_wit_Read_permission
```
Comment (azarovalex) [inline README.md:64]:
```suggestion
    password sk.ey_your_access_token_with_Read_permission
```

**Review by azarovalex — APPROVED** [review]
lgtm

File: Examples/mapbox_access_token (hunk 1 above)
Comment (friendoye) [reply Examples/mapbox_access_token:1]: 🫡

File: README.md (hunk 2 above)
Comment (aokj4ck) [reply README.md:64]: This is fixed on main 🤭 https://github.com/mapbox/mapbox-search-ios/blob/main/README.md?plain=1#L66

File: Examples/mapbox_access_token (hunk 1 above)
Comment (aokj4ck) [reply Examples/mapbox_access_token:1]: 🫡

**Pull request merged**